verification/src/contextual_block_verifier.rs

```diff
@@ -232,33 +232,16 @@ where
 
     pub fn verify(&self) -> Result<Vec<Capacity>, Error> {
         let cellbase = &self.resolved[0];
-        if (self.parent.number() + 1) > self.provider.consensus().reserve_number() {
-            let (target_lock, block_reward) = self
-                .provider
-                .finalize_block_reward(self.parent)
-                .map_err(|_| Error::CannotFetchBlockReward)?;
-            if cellbase.transaction.outputs_capacity()? != block_reward {
-                return Err(Error::Cellbase(CellbaseError::InvalidRewardAmount));
-            }
-            if cellbase.transaction.outputs()[0].lock != target_lock {
-                return Err(Error::Cellbase(CellbaseError::InvalidRewardTarget));
-            }
-        } else {
-            if cellbase.transaction.outputs_capacity()?
-                != self
-                    .provider
-                    .consensus()
-                    .genesis_epoch_ext()
-                    .block_reward(self.parent.number() + 1)
-                    .map_err(|_| Error::CannotFetchBlockReward)?
-            {
-                return Err(Error::Cellbase(CellbaseError::InvalidRewardAmount));
-            }
-            if &cellbase.transaction.outputs()[0].lock != self.provider.consensus().bootstrap_lock()
-            {
-                return Err(Error::Cellbase(CellbaseError::InvalidRewardTarget));
-            }
-        };
+        let (target_lock, block_reward) = self
+            .provider
+            .finalize_block_reward(self.parent)
+            .map_err(|_| Error::CannotFetchBlockReward)?;
+        if cellbase.transaction.outputs_capacity()? != block_reward {
+            return Err(Error::Cellbase(CellbaseError::InvalidRewardAmount));
+        }
+        if cellbase.transaction.outputs()[0].lock != target_lock {
+            return Err(Error::Cellbase(CellbaseError::InvalidRewardTarget));
+        }
```

Comment:
What if the miner construct a cellbase with two outputs whose sum equals `block_reward` but use a different lock on second output?

Reply:
There's a verification rule that cellbase can only have one input and one output.

Reply:
pub fn is_cellbase(&self) -> bool {
        self.inputs.len() == 1
            && self.outputs.len() == 1
            && self.witnesses.len() == 1
            && self.inputs[0].previous_output.is_null()
    }

we enforce `cellbase` have only one output.

Reply:
https://github.com/nervosnetwork/ckb/blob/develop/verification/src/block_verifier.rs#L50